src/position.h

```diff
@@ -1500,6 +1520,35 @@ inline bool Position::allow_virtual_drop(Color c, PieceType pt) const {
                         && count_in_hand(c, QUEEN) >= 0);
 }
 
+inline bool Position::sudoku_boxes() const {
+  assert(var != nullptr);
+  return var->sudoku && var->sudokuBoxWidth && var->sudokuBoxHeight;
+}
+
+inline int Position::sudoku_box_of(Square s) const {
+  assert(var != nullptr);
+  assert(sudoku_boxes());
+  return rank_of(s) / var->sudokuBoxHeight * (files() / var->sudokuBoxWidth) + file_of(s) / var->sudokuBoxWidth;
```

Comment:
you can use precalculated array instead of this, like `sudoku_box_of[to]`

Reply:
Right, but it's not a heavy calculation, and it depends on variant parameters, so I prefer not to over-complicate the code